src/api/entities/SecurityToken/CorporateActions/index.ts

```diff
@@ -104,32 +104,36 @@ export class CorporateActions extends Namespace<SecurityToken> {
    */
   public remove: ProcedureMethod<RemoveCorporateActionParams, void>;
 
-  // TODO @shuffledex
   /**
-   * Retrieve the Security Token's Corporate Actions agent
+   * Retrieve a list of agent identities
    */
-  public async getAgent(): Promise<Identity> {
+  public async getAgents(): Promise<Identity[]> {
     const {
       context: {
         polymeshApi: {
-          query: { corporateAction },
+          query: { externalAgents },
         },
       },
       parent: { ticker },
       context,
     } = this;
 
-    const rawTicker = stringToTicker(ticker, context);
+    const groupOfAgent = await externalAgents.groupOfAgent.entries(stringToTicker(ticker, context));
 
-    // const agent = await corporateAction.agent(rawTicker);
+    const agentIdentities: Identity[] = [];
 
-    // if (agent.isNone) {
-    const token = new SecurityToken({ ticker }, context);
-    const { owner } = await token.details();
-    return owner;
-    // }
+    groupOfAgent.forEach(([storageKey, agentGroup]) => {
+      if (agentGroup.isSome) {
```

Comment:
👍🏻

Reply:
I think it is possible, waiting for core answer yet to confirm. 
But I think it is possible to set a `none` AgentGroup to an agent. It is still an agent but has no permissions over the asset. 
I can't imagine the use case for it, but these checks are ok from our side, as it is possible to get empty groups here.

Reply:
I think I know the use case. Once an Identity is an Agent, even if they have no AgentGroup, you don't need to invite them again and to wait until they accept to give them an AgentGroup. 
So it could be a way to temporally remove permissions to agents.